src/mpid/ch4/netmod/ofi/subconfigure.m4

```diff
@@ -285,6 +285,18 @@ AM_COND_IF([BUILD_CH4_NETMOD_OFI],[
                                        [],
                                        [AC_MSG_ERROR(MPL_aligned_alloc is required to build OFI netmod)])
 
+    AC_ARG_ENABLE(ofi-domain,
```

Comment:
Why does it have to be a configure time option? Wouldn't it be useful to make it a runtime CVAR? Since it only affects initializing logic, I wouldn't expect too much runtime overhead for making it a CVAR.

Reply:
Maybe. But keep it a config option keeps the code more isolated and is more flexible. For example, it allows compile time logic. I think there's benefits to keep it simple during early development. We always can add infrastructure and enable runtime behavior later when we have better understanding.

Reply:
In my opinion CVAR is more useful in the early development phase, because that way we could switch between modes for experiments without rebuilding the whole library. Plus, this is also my opinion, but I would think `if (CVAR)` is simpler than `#ifdef XXX`.

Reply:
Moreover, particularly for this case I don't see a strong reason to make it compile time option (such as constant folding through in-lining).

Reply:
Yes, for now there isn't visible strong reasons against using CVAR and make it into runtime. It is also not visible that which direction toward VCI is more scalable and whether each may take some diverging paths such as different data structure, different stages of initialization and potential lazy schemes. Once the runtime infra-structure is in place, they potentially limit on new features or new experiments that we may try. Again, I am not saying I know runtime option is a bad idea, but at this point, the end direction is not clear, so I would like to keep it open. In the end, if there is clear winner, we probably will pick the winning choice and delete the other failed attempt. The runtime is reserved for the case that we remain ambiguous on what is the best solution and serve as a compromise. We always can add that option later.

Reply:
It is also possible that in the end there is not much diverging code between two approaches, which makes runtime choice remain simple, but I don't think we are sure about this possibility at this point yet.

Reply:
> but I would think `if (CVAR)` is simpler than `#ifdef XXX`.

I agree with you on this. However, I view the `#ifdef` during early stage of development as scaffold. It is ugly for serve a purpose. If they remain in the code after the code mature, that is an oversight on us. So we may convert to `if(CVAR)` later if that is clear, but not yet, in my opinion.